osquery/tables/system/windows/drivers.cpp

```diff
@@ -264,20 +265,17 @@ QueryData genDrivers(QueryContext& context) {
     }
 
     if (r.count("service") > 0 && !r.at("service").empty()) {
-      auto svc_key = kServiceKeyPath + r["service"];
-      r["service_key"] = svc_key;
-
-      // If the image map doesn't exist in the cache, manually look it up
-      auto svc_key_it = svc_image_map.find(svc_key);
-      if (svc_key_it != svc_image_map.end()) {
-        r["image"] = svc_key_it->second;
-      } else {
-        // Manual lookups of the service keys image path are _very_ slow
-        VLOG(1) << r["service"]
-                << " not found in image cache, performing manual lookup";
-        r["image"] = getDriverImagePath(svc_key);
+      std::string path;
+      r["service_key"] = kServiceKeyPath + r["service"];
```

Comment:
I think we should not make the assumption that the composition of these two paths is correct and points to a registry key.
I would first check that the registry key actually exists first.
If it does, then we set `service_key`, if it doesn't then we leave `service_key` empty and we don't attempt to extract the driver image path later.
Also we should log with VLOG if the `service_key` registry key doesn't exists.

A similar thing can be said for the `driver_key` at line 264. I would first check that the full path is actually a registry key, if not I would leave it empty.
Note that obviously in this case since we don't have a `driver` column or similar like in the `service` case we are "losing" an information that comes from the device properties, but I would prefer keeping consistency for now and make sure that `driver_key` is an actual registry key.

Reply:
Indeed, better be a bit cautious with registry path assumptions than the retrieved data will wrong.